Assets/Scripts/SaveSystem/testClasses/TestEntity.cs

```diff
@@ -0,0 +1,50 @@
+﻿using UnityEngine;
+using SaveSystem;
+
+namespace KainsTestScripts
+{
+    //Example of a class/component that implements ISaveable and is saved to the SaveSystem
+    public class TestEntity : MonoBehaviour, ISaveableComponent
+    {
+
+        public float health = 100;
+
+        void Start()
+        {
+            //im a normal script that someone made. look at me gooo
+
+        }
+
+
+
+
+
+
+        #region SAVE SYSTEM
+        [System.Serializable]
+        protected class TestEntitySaveData : SaveData //class that is a container for data that will be saved
```

Comment:
So each ISavableComponent needs to create a subclass with SaveData?

Reply:
yes, i wonder if theres a better way though